sv.h

```diff
@@ -1902,19 +1902,22 @@ END_EXTERN_C
     (((SvFLAGS(sv) & (SVf_POK|SVs_GMG)) == SVf_POK) || ((SvFLAGS(sv) & (SVf_IOK|SVp_POK|SVs_GMG)) == (SVf_IOK|SVp_POK)))
 
 #define SvPV_flags(sv, len, flags)                                          \
+   (__ASSERT_(sizeof(len) == sizeof(STRLEN))                                \
```

Comment:
Why `__ASSERT_()` (which really isn't a name we should define, let alone use) instead of `assert(...),`?

Unfortunately we can't use a static assert here :(

Reply:
Bram++  pointed me to a discussion about using a static assert expression.  I had a go at that, and am stymied.

Yes, one can create such a thing for C, but apparently not for C++, without using lambda expressions.  But I don't consider this a show stopper.  For C++ compilations, just expand to nothing

The problem I run into is many compilation warnings because we enable C++ compat warnings.  AFAICT the pragmas to turn those off around these are statements, not an expression